src/components/Item.jsx

```diff
@@ -0,0 +1,34 @@
+import React from 'react'
+import { useState } from 'react'
+import '../App.css'
+
+export default function Item({ item, onChange, onDelete }) {
+    const[updating, setUpdating] = useState(false)
+
+    let itemInfo
+    if(updating){
+        itemInfo=(
+            <>
+        <input type='text' value={item.text} aria-label='edit-input' name='edit-input'  onChange={(e) => {onChange({ ...item, text: e.target.value, })}
```

Comment:
it would be better to have a text piece of state that the onChange updates instead of updating state with every letter the user types.